strongbox-user-management/src/main/java/org/carlspring/strongbox/authentication/registry/support/AuthenticatorsScanner.java

```diff
@@ -0,0 +1,113 @@
+package org.carlspring.strongbox.authentication.registry.support;
+
+import org.carlspring.strongbox.authentication.api.Authenticator;
+import org.carlspring.strongbox.authentication.registry.AuthenticatorsRegistry;
+import org.carlspring.strongbox.authentication.registry.support.xml.XmlAuthenticator;
+import org.carlspring.strongbox.authentication.registry.support.xml.XmlAuthenticators;
+import org.carlspring.strongbox.resource.ConfigurationResourceResolver;
+import org.carlspring.strongbox.xml.parsers.GenericParser;
+
+import javax.inject.Inject;
+import java.io.IOException;
+import java.util.List;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import com.google.common.base.Throwables;
+import org.apache.commons.lang.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.support.BeanDefinitionRegistry;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.annotation.AnnotationConfigRegistry;
+import org.springframework.context.annotation.ClassPathBeanDefinitionScanner;
+import org.springframework.core.io.Resource;
+
+/**
+ * @author Przemyslaw Fusik
+ */
+public class AuthenticatorsScanner
```

Comment:
I think it's not the same things with `BeanFactoryPostProcessor `, i think that in your case it will be better for `Authenticator` to provide it's own application context thru applicationContext.xml or `@Configuration`, which must be nesteed for strongbox application context. In this case these nesteed context will be isolated with each other.